webview-ui/src/components/history/HistoryPreview.tsx

```diff
@@ -2,99 +2,117 @@ import { memo } from "react"
-				<Checkbox
-					id="show-all-workspaces"
-					checked={showAllWorkspaces}
-					onCheckedChange={(checked) => setShowAllWorkspaces(checked === true)}
-					variant="description"
-				/>
-				<label htmlFor="show-all-workspaces" className="text-xs text-vscode-foreground cursor-pointer">
-					{t("history:showAllWorkspaces")}
-				</label>
-			</div>
-			{tasks.slice(0, 3).map((item) => (
-				<div
-					key={item.id}
-					className="bg-vscode-toolbar-hoverBackground/50 hover:bg-vscode-toolbar-hoverBackground/75 rounded-xs relative overflow-hidden opacity-90 hover:opacity-100 cursor-pointer"
-					onClick={() => vscode.postMessage({ type: "showTaskWithId", text: item.id })}>
-					<div className="flex flex-col gap-2 p-3 pt-1">
-						<div className="flex justify-between items-center">
-							<span className="text-xs font-medium text-vscode-descriptionForeground uppercase">
-								{formatDate(item.ts)}
-							</span>
-							<CopyButton itemTask={item.task} />
+		<>
+			<div className="flex flex-col gap-3 shrink-0 mx-5">
+				{!!tasks.length && (
+					<div className="flex items-center justify-between text-vscode-descriptionForeground">
+						<div className="flex items-center gap-1">
+							<span className="codicon codicon-comment-discussion scale-90 mr-1" />
+							<span className="font-medium text-xs uppercase">{t("history:recentTasks")}</span>
 						</div>
-						<div
-							className="text-vscode-descriptionForeground overflow-hidden whitespace-pre-wrap"
-							style={{
-								display: "-webkit-box",
-								WebkitLineClamp: 3,
-								WebkitBoxOrient: "vertical",
-								wordBreak: "break-word",
-								overflowWrap: "anywhere",
-							}}>
-							{item.task}
-						</div>
-						<div className="text-xs text-vscode-descriptionForeground">
-							<span>
-								{t("history:tokens", {
-									in: formatLargeNumber(item.tokensIn || 0),
-									out: formatLargeNumber(item.tokensOut || 0),
-								})}
-							</span>
-							{!!item.cacheWrites && (
-								<>
-									{" • "}
-									<span>
-										{t("history:cache", {
-											writes: formatLargeNumber(item.cacheWrites || 0),
-											reads: formatLargeNumber(item.cacheReads || 0),
-										})}
-									</span>
-								</>
-							)}
-							{!!item.totalCost && (
-								<>
-									{" • "}
-									<span>{t("history:apiCost", { cost: item.totalCost?.toFixed(4) })}</span>
-								</>
+						<Button variant="ghost" size="sm" onClick={() => showHistoryView()} className="uppercase">
+							{t("history:viewAll")}
+						</Button>
+					</div>
+				)}
+				{tasks.length === 0 && (
+					<>
+						<p className="outline rounded p-4">
+							<Trans
+								i18nKey="chat:onboarding"
+								components={{
+									DocsLink: (
+										<a
+											href="https://docs.roocode.com/getting-started/your-first-task"
+											target="_blank"
+											rel="noopener noreferrer">
+											the docs
```

Comment:
Probably don't want this english string here

Reply:
This gets replaced in i18n, and it is technically the way `<Trans>` is supposed to work. Typescript/linter complains if you have an empty element here, so it has to stay.

Reply:
Ahh got it, thanks!